### PR TITLE
TextInput with icon

### PR DESCRIPTION
**Pull request by mperrotti**
#### Related issues
Fixes https://meetup.atlassian.net/browse/SDS-432

#### Description
Adds the option to have text inputs with icons. Chose to use a `background-image` instead of `<Icon>` in order to have the least changes to `<TextInput>` as possible.

#### Screenshots (if applicable)
screen shot 2017-10-10 at 4 14 16 pm

### CODE REVIEW

File: src/forms/textInput.test.jsx [hunk 1]
```diff
@@ -112,5 +117,11 @@ describe('TextInput', function() {
 			const errorEl = component.find('.text--error').getDOMNode();
 			expect(errorEl.textContent).toEqual(ERROR_TEXT);
 		});
+
+		// it('should show an icon when iconShape prop is specified', () => {
```
Comment (emilyeng) [inline src/forms/textInput.test.jsx]: can this be removed?

File: src/forms/TextInput.jsx [hunk 2]
```diff
@@ -38,13 +45,26 @@ const TextInput = (props) => {
 		labelClassName
 	);
 
+	const iconSize = props.iconSize || 'xs';
```
Comment (emilyeng) [inline src/forms/TextInput.jsx]: you can pull `iconSize` out above with the other props (line ~29)

File: src/forms/textInput.test.jsx (hunk 1 above)
Comment (mperrotti) [reply src/forms/textInput.test.jsx]: Yea, my mistake

File: src/forms/TextInput.jsx (hunk 2 above)
Comment (mperrotti) [reply src/forms/TextInput.jsx]: Keeping as-is in the interest of not having to define a new variable for icon size

**Pull request merged**